consensus/polybft/event_tracker.go

```diff
@@ -0,0 +1,79 @@
+package polybft
+
+import (
+	"context"
+	"path/filepath"
+
+	hcf "github.com/hashicorp/go-hclog"
+	"github.com/umbracle/ethgo"
+	"github.com/umbracle/ethgo/jsonrpc"
+	"github.com/umbracle/ethgo/tracker"
+	boltdbStore "github.com/umbracle/ethgo/tracker/store/boltdb"
+)
+
+type eventSubscription interface {
+	AddLog(log *ethgo.Log)
+}
+
+type eventTracker struct {
+	dataDir    string
+	config     *PolyBFTConfig
+	subscriber eventSubscription
+	logger     hcf.Logger
+}
+
+func (e *eventTracker) start() error {
+	provider, err := jsonrpc.NewClient(e.config.Bridge.JSONRPCEndpoint)
+	if err != nil {
+		return err
+	}
+
+	store, err := boltdbStore.New(filepath.Join(e.dataDir, "/deposit.db"))
+
+	if err != nil {
+		return err
+	}
+
+	e.logger.Info("Start tracking events", "bridge", e.config.Bridge.BridgeAddr)
+
+	tt, err := tracker.NewTracker(provider.Eth(),
+		tracker.WithBatchSize(10),
+		tracker.WithStore(store),
+		tracker.WithFilter(&tracker.FilterConfig{
+			Async: false,
+			Address: []ethgo.Address{
+				ethgo.Address(e.config.Bridge.BridgeAddr),
+			},
+		}),
+	)
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		go func() {
+			if err := tt.Sync(context.Background()); err != nil {
+				e.logger.Error("Event tracker", "failed to sync", err)
+			}
+		}()
+
+		go func() {
+			for {
+				select {
+				case evnt := <-tt.EventCh:
+					if len(evnt.Removed) != 0 {
+						panic("this will not happen anymore after tracker v2")
```

Comment:
I believe we need a `tech-debt` label in Jira or something. I believe we eventually need to replace panics with error handling.

Reply:
I don't fully agree with this - people shouldn't be creating tech debt in the first place under the premise that it will be resolved in the future (it won't).

This entire segment needs to be refactored now